packages/block-library/src/unsupported-block-modal/index.native.js

```diff
@@ -0,0 +1,140 @@
+				const wpToolbar = document.getElementById( 'wp-toolbar' );
+				if ( wpAdminBar ) {
+					wpAdminBar.style.display = 'none';
+				}
+				if ( wpToolbar ) {
+					wpToolbar.style.display = 'none';
+				}
+			
+				const content = document.getElementById( 'wpbody-content' );
+				if ( content ) {
+					const callback = function( mutationsList, observer ) {
+						const header = document.getElementsByClassName(
+							'edit-post-header'
+						)[ 0 ];
+						const postTitle = document.getElementById( 'post-title-0' );
+						if ( postTitle && header.id === '' ) {
+							header.id = 'gb-header';
+							header.style.height = 0;
+							postTitle.style.display = 'none';
+							Array.from( header.children ).forEach( ( child ) => {
+								child.style.display = 'none';
+							} );
+			
+							const headerToolbar = header.getElementsByClassName(
+								'edit-post-header-toolbar'
+							)[ 0 ];
+							headerToolbar.style.display = null;
+							headerToolbar.parentNode.style.display = null;
+							const inserterToggle = header.getElementsByClassName(
+								'block-editor-inserter__toggle'
+							)[ 0 ];
+							inserterToggle.style.display = 'none';
+			
+							const blockToolbar = header.getElementsByClassName(
+								'edit-post-header-toolbar__block-toolbar'
+							)[ 0 ];
+							blockToolbar.style.top = '0px';
+			
+							// const skeleton = document.getElementsByClassName(
+							// 	'block-editor-editor-skeleton'
+							// )[ 0 ];
+							//                skeleton.style.top = '0px';
+			
+							const appender = document.getElementsByClassName(
+								'block-list-appender'
+							)[ 0 ];
+							if ( appender && appender.id === '' ) {
+								appender.id = 'appender';
+								appender.style.display = 'none';
+							}
+			
+							window.insertBlock();
+							observer.disconnect();
+						}
+					};
+					/* eslint-disable-next-line no-undef */
+					const observer = new MutationObserver( callback );
+			
+					const config = { attributes: true, childList: true, subtree: true };
+					observer.observe( content, config );
+				}
+			};
+			
+			/* eslint-disable-next-line no-unused-expressions */
+			true; // react-native-webview asks for it			
+		`;
+
+		return (
+			<Modal
+				animationType="slide"
+				presentationStyle="fullScreen"
+				visible={ this.props.visible }
+				onRequestClose={ () => this.props.closeModal() }
+			>
+				<SafeAreaView style={ { flex: 1, backgroundColor: 'black' } }>
+					<Button
+						onPress={ () => this.props.closeModal() }
+						title="Close"
+						color="red"
+					/>
+					<Button
+						onPress={ () =>
+							this.webref.injectJavaScript( 'window.onSave()' )
+						}
+						title="Save"
+						color="red"
+					/>
+					<WebView
+						ref={ ( r ) => ( this.webref = r ) }
+						source={ {
+							uri:
+								'https://cruisinglamb.wordpress.com/wp-admin/post-new.php',
+						} }
+						injectedJavaScript={ scriptToInject }
```

Comment:
Do we have more control over when can we inject JS code?

I have noticed that to avoid some flickering of UI or CSS overrides we will need to inject some snippets as early as possible and others as late as possible.

There's for example this little flash of the WP Admin Bar while the page is loading. So we should inject the "hide WP Admin Bar" CSS earlier. But then the Editor CSS will override the Gutenberg top bar changes since it loads later. And to avoid that, we need to inject the Editor CSS as late as possible (when everything is loaded).

![flash](https://user-images.githubusercontent.com/9772967/80131520-a8b34480-859a-11ea-8c66-538e99e20e6a.gif)